src/App.tsx

```diff
@@ -1,24 +1,270 @@
+        duration: 500,
+        position,
+      });
+      break;
+    default:
+      toast(message);
+  }
+};
+
+export const App: FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [tempTodo, setTempTodo] = useState<Todo | null>(null);
+  const [filterType, setFilterType] = useState(FilterType.All);
+  const [isInputDisabled, setIsInputDisabled] = useState(false);
+  const [loadingTodosIds, setLoadingTodosIds] = useState<number[]>([]);
+
+  const getCompletedTodos = () => todos.filter(({ completed }) => completed);
+  const getUnCompletedTodos = () => todos.filter(({ completed }) => !completed);
+
+  const [activeTodosCount, completedTodosCount] = useMemo(
+    () => [getUnCompletedTodos().length, getCompletedTodos().length],
+    [todos],
+  );
+
+  useEffect(() => {
+    notify('Loading todos...', NotificationType.Loading);
+
+    getTodos(USER_ID)
+      .then((fetchedTodos) => {
+        setTodos(fetchedTodos);
+        notify('Successfully loaded todos!', NotificationType.Success);
+      })
+      .catch(() => {
+        notify('Unable to load todos!', NotificationType.Error);
+      });
+  }, []);
+
+  const handleAddTodo = async (title: string) => {
+    setIsInputDisabled(true);
+
+    if (!title.length) {
+      notify("Title can't be empty!", NotificationType.Error);
+    } else {
+      notify('Adding new todo...', NotificationType.Loading);
+      const newTodo = {
+        id: 0,
+        userId: USER_ID,
+        completed: false,
+        title,
+      };
+
+      setTempTodo(newTodo);
+      setLoadingTodosIds((prev: number[]) => [...prev, newTodo.id]);
+
+      try {
+        const postedTodo = await postTodo(newTodo);
+
+        setTodos((prevTodos) => {
+          notify('Successfully added new todo!', NotificationType.Success);
+
+          return [...prevTodos, postedTodo];
+        });
+      } catch {
+        notify('Unable to add a todo!', NotificationType.Error);
+      } finally {
+        setTempTodo(null);
+        setLoadingTodosIds([]);
+      }
+    }
+
+    setIsInputDisabled(false);
+  };
+
+  const handleDeleteTodo = useCallback(
+    async (id: number) => {
+      setLoadingTodosIds((prev: number[]) => [...prev, id]);
+      notify('Deleting todo...', NotificationType.Loading);
+
+      try {
+        const deletedTodo = await deleteTodo(id);
+
+        setTodos((prevTodos) => {
+          notify('Successfully deleted a todo!', NotificationType.Success);
+
+          return prevTodos.filter((todo) => todo.id !== id);
+        });
+
+        return deletedTodo;
+      } catch (error) {
+        notify('Unable to delete a todo!', NotificationType.Error);
+      } finally {
+        setTempTodo(null);
+        setLoadingTodosIds([]);
+      }
+
+      return null;
+    },
+    [deleteTodo],
+  );
+
+  const handleDeleteCompleted = () => {
+    getCompletedTodos().forEach(({ id }) => {
+      handleDeleteTodo(id).then(() => setTodos(getUnCompletedTodos()));
```

Comment:
you already set todos in your deleteTodo method
```suggestion
      handleDeleteTodo(id)
```